app/views/courses/moss.html.erb

```diff
@@ -1,177 +1,166 @@
+																																				 ["FORTRAN", "fortran"],
+																																				 ["Haskell", "haskell"],
+																																				 ["Java", "java"],
+																																				 ["Javascript", "javascript"],
+																																				 ["Lisp", "lisp"],
+																																				 ["MIPS Assembly", "mips"],
+																																				 ["ML", "ml"],
+																																				 ["Matlab", "matlab"],
+																																				 ["Modula2", "modula2"],
+																																				 ["Pascal", "pascal"],
+																																				 ["Perl", "perl"],
+																																				 ["PL/SQL", "plsql"],
+																																				 ["Prolog", "prolog"],
+																																				 ["Python", "python"],
+																																				 ["Scheme", "scheme"],
+																																				 ["Spice", "spice"],
+																																				 ["VHDL", "vhdl"],
+																																				 ["Verilog", "verilog"],
+																																				 ["Visual Basic", "vb"],
+																																				 ["a8086 Assembly", "a8086"]
+																																			 ], "c") %>
+								<label>Language Options</label>
+							</div>
 						</div>
+					</li>
+					<li>
+						<label>
+							<%= check_box_tag "box_max","1",false, :class => "filled-in" %>
+							<span>Max Lines </span>
+						</label>
 						<br>
-						<i>This if used must be an non-nested TAR. Contents of which will not be run against the assignments. Perfect for example code, or stubbed out functions that were given out.</i>
-					</div>
-				</li>
-			</ul>
-    </li>
-  </ul>
+						<div style="padding-left:10px;" >
+							<%= number_field_tag "max_lines", "10", placeholder: "If you don't know what this is please read the Moss documentation before altering this value.", min: 1 %>
+						</div>
+					</li>
+					<li>
+						<label>
+							<%= check_box_tag "box_basefile", "1",false, :class => "filled-in" %>
+							<span>Base File Tar </span>
+						</label>
+						<div style="padding-left:10px;" >
+							<div class="file-field input-field">
+								<div class="btn">
+									<span>Base File</span>
+									<%= file_field_tag 'base_tar' %>
+								</div>
+								<div class="file-path-wrapper">
+									<input class="file-path validate" type="text" placeholder="Additional files archive">
+								</div>
+							</div>
+							<br>
+							<i>This if used must be an non-nested TAR. Contents of which will not be run against the assignments. Perfect for example code, or stubbed out functions that were given out.</i>
+						</div>
+					</li>
+				</ul>
+			</li>
+		</ul>
+
+		<br>
 
-  <br>
+		<h4>Step 3:</h4>
+		<p> (Optional) Upload an archive containing additional files you'd like Moss to compare against.</p>
+		<p>(Note: the archive must contain only regular files.  Nested archives wil not be extracted.)</p>
```

Comment:
One last thing -- I realized that there's a typo "wil" here. While we're on the topic of MOSS, I figured this might be a good time to fix it.